Adds "Try it out" section for the simple GKE example.

psx95 opened the pull request:
Fixes #20 

### Description
Adds "Try it out" section for the simple GKE example. 

The section explains how to use JSON files to send metrics, logs and traces to the Collector running in a GKE cluster. 

### Testing
Followed the instructions in the README to successfully emit metrics, traces and logs to the stdout and google cloud.

psx95 committed:
Update build configs to generate collectors with jsonfilereceiver

psx95 committed:
Add 'try it out' section for redaction sample

This allows the user to see redaction sample work on custom
user-provided telemetry data.

psx95 committed:
Removes busybox from Dockerfile

psx95 committed:
Updates instructions with better language

psx95 committed:
Removes extra newline

psx95 committed:
Adds missing language fixes

psx95 committed:
Adds missing newline at EOF

psx95 committed:
Adds sources for sample telemetry data

psx95 committed:
Adds completed configuration for collector

psx95 committed:
Adds 'try-it-out' section for simple example

damemi commented on deploy/gke/simple/README.md:
```diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+
+### Updating the collector configuration to receive data from JSON file
+
+We need to update the collector configuration file - [otel-config](./otel-config.yaml) to add a reciever that is able to receive telemetry data from the added JSON file.
```
```suggestion
We need to update the collector configuration [otel-config.yaml](./otel-config.yaml) file to add a reciever that is able to receive telemetry data from the added JSON file.
```

damemi commented on deploy/gke/simple/README.md:
````diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+    exporters: [googlecloud, logging]
+```
+
+For exporting logs to Google Cloud Platform, we need to further configure the collector with the log name and the GCP project ID. The project ID is used by the cloud exporter to create GCP log entries. 
````
is project_id not inferred by the collector based on the GKE node it's running on?

psx95 replied:
No, it was not since the logs were not being generated. I can double check again though by removing it and keeping the `default_log_name`.

psx95 replied:
Just double checked, turns out that `default_log_name` was the required field in this case. Project ID is inferred.

damemi commented on deploy/gke/simple/README.md:
````diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+        - name: test-data-vol-logs  # This name should match to the name of volume in spec.volumes
+          mountPath: /mnt/testdata/logs   # This is the location in the kubernetes environment where the file for configMap for logs will be mounted
+        ```
+6. Deploy the cluster using the new deployment manifest - 
````
```suggestion
6. Deploy the collector using the new deployment manifest - 
```

damemi commented on deploy/gke/simple/README.md:
```diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+
+*If the logs do not show the expected telemetry data, check if the file used to send json data - [testdata.json](./otlp-data/testdata.json) is found in the Kubernetes deployment. Follow steps in [troubleshooting guide](../troubleshooting.md#verify-updates-to-filesconfigmaps-in-kubernetes-cluster) to verify file presence and contents.*
+
+**NOTE: You will need to update the timestamps in [testdata-metrics](./otlp-data/testdata-metrics.json) to be within 24 hours of current time otherwise the metrics will not show up in Google Cloud console's Metrics Explorer. You might also need to update timestamps in other testdata - metrics and logs in case they become old enough to not be recognized by Google Cloud.**
```
is there any way we could automatically update these for the users with a Make command or something? Or give the user a time window to look at in their metrics explorer?

psx95 replied:
I think we could use `sed` with `make` to automate this. Sounds like a good idea.

damemi commented on deploy/gke/simple/README.md:
````diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+    kubectl logs <pod_container_name> -n $OTEL_NAMESPACE
+    ```
+
+*If the logs do not show the expected telemetry data, check if the file used to send json data - [testdata.json](./otlp-data/testdata.json) is found in the Kubernetes deployment. Follow steps in [troubleshooting guide](../troubleshooting.md#verify-updates-to-filesconfigmaps-in-kubernetes-cluster) to verify file presence and contents.*
````
I would cut this to just the last sentence
```
Follow steps in [troubleshooting guide](../troubleshooting.md#verify-updates-to-filesconfigmaps-in-kubernetes-cluster) to verify file presence and contents.
```

psx95 replied:
Right, I will shorten it. I would prepend it with "In case there are any issues, " otherwise it might appear as a mandatory step.

damemi commented on deploy/gke/simple/README.md:
```diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+
+### Seeing the telemetry data
+
+At this point, we have several files acting as sources of telemetry data and a reciever configured withing the OpenTelemetry Collector that listens to these files whenever there are updates made to them. The collector also receives telemetry data from the files once when they are initially addedd (mounted).
```
```suggestion
At this point, we have several files acting as sources of telemetry data and a receiver configured within the OpenTelemetry Collector that listens to these files whenever there are updates made to them. The collector also receives telemetry data from the files once when they are initially mounted.
```

damemi commented on deploy/gke/simple/README.md:
````diff
@@ -67,6 +67,171 @@ Since there is no application currently running on the cluster, the collector is
+
+To check if the test file at mount path is updated, use following command with correct file path - 
+```
+ kubectl exec -n $OTEL_NAMESPACE -it  <pod_container_name> -- cat /mnt/testdata/traces/testdata-traces.json 
````
this will only work if they're following the troubleshooting doc right? Maybe it should be moved there

psx95 replied:
Yes, it does seem similar to the toubleshooting step but this is different from troubleshooting. 

This is in case the user wants to update their JSON data. Like it is possible that user reaches this step without needing to open troubleshooting, but I can remove this and link the troubleshooting section since content is same.

damemi replied:
Yeah, the problem is that this command won't work unless they do the steps to update the Dockerfile and give them a shell in the container

psx95 replied:
Oh yes, I see your point. I agree, better to move it there.

psx95 committed:
Update deploy/gke/simple/README.md

Co-authored-by: Mike Dame <mikedame@google.com>

psx95 committed:
Update deploy/gke/simple/README.md

L185

Co-authored-by: Mike Dame <mikedame@google.com>

psx95 committed:
Update deploy/gke/simple/README.md

L204

Co-authored-by: Mike Dame <mikedame@google.com>

psx95 committed:
Update README

psx95 committed:
Add make command to automate timestamp update

psx95 committed:
Removes explicit declaration of Project ID from googlecloud exporter config

The default project ID is inferred and so this field is only required
in case telemetry data needs to be put in a different project that what
is set as the default GCP project
